disperser/common/blobstore/blob_metadata_store.go

```diff
@@ -526,10 +533,14 @@ func GenerateTableSchema(metadataTableName string, readCapacityUnits int64, writ
 	}
 }
 
+// MarshalBlobMetadata marshals the blob metadata into a DynamoDB item, or else returns a grpc internal error
 func MarshalBlobMetadata(metadata *disperser.BlobMetadata) (commondynamodb.Item, error) {
+	// We use MarshalMap a few times here. Errors from MarshalMap are most likely
+	// a bug in our code (badly set configurations), so not worth client retrying,
+	// hence we send a 500s instead of 503s
```

Comment:
Not sure this comment section is necessary and it seems kinda passive. An error here definitely indicates a bug in our code. The idea of returning 503s here is the wrong mental model for 503s. So adding `instead of 503s` seems like an odd thing to bring up.

Reply:
My idea was that if there’s a bug in our code, then sending a 503 and asking the client to resubmit won’t help; they will just hit the same bug again with the next request. 

This is why this separation between 500s and 503s feels so hard to me. Because im never sure if the error is not actually triggered by some one in a million scenario which wouldn’t happen if they retry for whatever reason. Maybe it is the case (then should send 503), and maybe it isn’t (then should send 500 to prevent them from spamming us with more requests)

Reply:
Ahhhh I see what you are saying now! You're saying that there's no point in even comparing to 503 because 503 is "clearly" wrong. I guess that's true.. I was stuck in this 503 vs 500 model while coding this PR, hence the comment. Will remove.